llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -20543,6 +20531,121 @@ unsigned RISCVTargetLowering::getMinimumJumpTableEntries() const {
   return Subtarget.getMinimumJumpTableEntries();
 }
 
+void RVVArgDispatcher::constructArgInfos(Type *Ty) {
```

Comment:
This seems to be reinventing splitToValueTypes

Reply:
It's a bit different since `splitToValueTypes` uses `ComputeValueVTs` to get value type(s) and it can't handle **vector tuple type**. Same situation in `SelectionDAGISel::LowerArguments`, they both use `ComputeValueVTs` for getting value types, that's why `RVVArgDispatcher` comes to help.

Reply:
What is a "vector tuple type"? The DAG doesn't seem to rely on a similar helper

Reply:
By **vector tuple type**, I mean RVV tuple type, for example: `vint32m2x2_t` which needs 2 lmul2 vector registers. The llvm models it as `struct {vint32m2_t, vint32m2_t}` and it would be flattened by backend which makes it impossible to differentiate normal vector type and vector tuple type in `AssignFn`, that's why I added the helper to both GISel and normal selection DAG.